src/core/currency/wallet/currency-wallet-api.ts

```diff
@@ -701,6 +704,18 @@ export function combineTxWithFile(
       }
     }
 
+    const mergedTokens: TransactionFile['tokens']['tokenId'] = mergeDeeply(
+      file.tokens[PARENT_TOKEN_ID],
+      file.tokens[tokenId ?? PARENT_TOKEN_ID]
+    )
+
+    if (mergedTokens.savedAction != null) {
+      out.savedAction = {
+        ...out.savedAction,
+        ...mergedTokens.savedAction
+      }
+    }
```

Comment:
I think this whole block could just be `out.savedAction = mergedTokens.savedAction`. There is no way `out.savedAction` can be anything other than `undefined`, if you follow the logic in this function.

I see similar logic above for the metadata, which is probably also a mistake. At least with metadata, though, you can always merge two structures together and get a valid metadata. With transaction actions, though, merging two together will usually produce an invalid value.